### PR TITLE
Add Support for PySpark

### PR DESCRIPTION
**Pull request by ryan-clancy**
Adds the Java classes required for initial PySpark support.

### CODE REVIEW

**Comment by ryan-clancy** [comment]
@lintool Minimal PySpark example is here

File: src/main/java/io/anserini/spark/JavaLuceneRDD.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class JavaLuceneRDD extends JavaRDD<Integer> {
+
+  public JavaLuceneRDD(JavaRDD<Integer> rdd) {
+    super(JavaRDD.toRDD(rdd), ClassManifestFactory.fromClass(Integer.class));
+  }
+
+  public JavaRDD<HashMap<String, String>> getDocs(String path) {
+
+    return wrapRDD(rdd()).mapPartitions((part) -> {
+
```
Comment (lintool) [inline src/main/java/io/anserini/spark/JavaLuceneRDD.java:27]: seems like there are two many blank lines?

File: src/main/java/io/anserini/spark/JavaLuceneRDD.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      // IndexReader per partition
+      IndexReader reader = DirectoryReader.open(new HdfsReadOnlyDirectory(new Configuration(), new Path(path)));
+
+      // Convert to Iterable
+      Iterable<Integer> iterable = () -> part;
+
+      return StreamSupport.stream(iterable.spliterator(), false).map(doc -> {
+
+        HashMap<String, String> fields = new HashMap();
```
Comment (lintool) [inline src/main/java/io/anserini/spark/JavaLuceneRDD.java]: Probably just a `Map<String, String>` in the first part?

**Comment by lintool** [comment]
@r-clancy Minor comments. lg!

Something we should think about moving forward - do we concurrently support Scala and Java? If so, how do we keep the abstractions in sync? Better testing? Or we just abandon Scala and support PySpark only via the Java bindings?

File: README.md [hunk 3]
```diff
@@ -26,3 +26,34 @@ docs: org.apache.spark.rdd.RDD[String] = MapPartitionsRDD[1]
...
+$ pyspark --jars target/anserini-spark-0.0.1-SNAPSHOT-fatjar.jar
+
+# Import Java -> Python converter
+from pyspark.mllib.common import _java2py
+
+# The path of the Lucene index
+INDEX_PATH = "/tuna1/indexes/lucene-index.core17.pos+docvectors+rawdocs/"
+
+# The JavaIndexLoader instance
+index_loader = sc._jvm.io.anserini.spark.JavaIndexLoader(sc._jsc, INDEX_PATH)
```
Comment (lintool) [inline README.md:44]: We can deal with this later in a separate PR - but maybe we should have some wrapper classes on the Python side so we don't have to deal with this `sc._jvm` ugliness... I believe this is what PySpark does, right?

**Comment by ryan-clancy** [comment]
@lintool Another option would be to use [PyLucene](https://lucene.apache.org/pylucene/) which is distributed by the Lucene team each release (each release consists of a Lucene + Solr + PyLucene version). I'm not sure how nicely it would play with Spark, but it might be worth looking into so we don't have to do the janky `sc._jvm` stuff ourselves - they expose an API that's very similar to the Java one we're used to.

We could probably wrap the Java code into the Scala versions of the Spark objects pretty easily, then the logic would be in Java and we just need to make sure we have the same methods exposed in Scala. Or, maybe the Java code could be called directly without any modifications? Lots of options and stuff to look into... maybe we can chat f2f about what approach we want to take and what's in-scope for the project vs. later work?

**Pull request merged**